src/main/java/org/shanerx/tradeshop/Potions.java

```diff
@@ -0,0 +1,299 @@
+    L_WEAK("weakness", "lingeringotion", false, false),
+    L_WEAK_T("weakness", "lingeringotion", true, false),
+    L_LUCK("luck", "lingeringotion", false, false);
+
+    private String potionType, type;
+    private boolean extended, amplified;
+
+    Potions(String p, String t, boolean e, boolean a) {
+        potionType = p;
+        type = t;
+        extended = e;
+        amplified = a;
+    }
+
+    public static String findPotion(ItemStack itm) {
+        if (itm.getType() != Material.POTION && itm.getType() != Material.SPLASH_POTION && itm.getType() != Material.LINGERING_POTION) {
+            return null;
+        }
+
+        PotionData pd = ((PotionMeta) itm.getItemMeta()).getBasePotionData();
+        String pT = "", t = "", mod = "", pot;
+
+        if (itm.getType() == Material.POTION) {
+            t = "P_";
+        } else if (itm.getType() == Material.SPLASH_POTION) {
+            t = "S_";
+        } else if (itm.getType() == Material.LINGERING_POTION) {
+            t = "L_";
+        }
+
+        if (pd.isExtended()) {
+            mod = "_T";
+        } else if (pd.isUpgraded()) {
+            mod = "_I";
+        }
+
+        switch (pd.getType().toString().toLowerCase()) {
+            case "uncraftable":
+                pT = "EMPTY";
+                break;
+            case "water":
+                pT = "WATER";
+                break;
+            case "thick":
+                pT = "THICK";
+                break;
+            case "awkward":
+                pT = "AWKWARD";
+                break;
+            case "night_vision":
+                pT = "NIGHT";
+                break;
+            case "invisibility":
+                pT = "INVIS";
+                break;
+            case "jump":
+                pT = "LEAP";
+                break;
+            case "speed":
+                pT = "SWIFT";
+                break;
+            case "slowness":
+                pT = "SLOW";
+                break;
+            case "water_breathing":
+                pT = "BREATH";
+                break;
+            case "mundane":
+                pT = "MUNDANE";
+                break;
+            case "fire_resistance":
+                pT = "FIRERES";
+                break;
+            case "instant_heal":
+                pT = "HEALTH";
+                break;
+            case "instant_damage":
+                pT = "HARM";
+                break;
+            case "poison":
+                pT = "POISON";
+                break;
+            case "regen":
+                pT = "REGEN";
+                break;
+            case "strength":
+                pT = "STRENGTH";
+                break;
+            case "weakness":
+                pT = "WEAK";
+                break;
+            case "luck":
+                pT = "LUCK";
+                break;
+        }
+
+        pot = t + pT + mod;
+
+        return pot;
+    }
+
+    public static Potions getPotion(String str) {
```

Comment:
What is the intention behind wrapping the Potions.valueOf() function? Simply renaming the method?

Reply:
isType checks for null and IllegalArguments which prevents errors, this returns the enum vs isType which is only a boolean